integration/kubernetes/k8s-expose-ip.bats

```diff
@@ -30,33 +29,26 @@ setup() {
 	cmd="kubectl wait --for=condition=Available deployment/${deployment}"
 	waitForProcess "$wait_time" "$sleep_time" "$cmd"
 
-	# Get deployment
-	kubectl get deployments ${deployment}
-
-	# Describe deployment
-	kubectl describe deployments ${deployment}
-
-	# Display information about ReplicaSet objects
-	kubectl get replicasets
-	kubectl describe replicasets
+	# Check pods are running
+	cmd="kubectl get pods -o jsonpath='{.items[*].status.phase}' | grep Running"
+	waitForProcess "$wait_time" "$sleep_time" "$cmd"
 
 	# Expose deployment
 	kubectl expose deployment/${deployment} --type=LoadBalancer --name=${service}
 
-	# Get service
-	kubectl get services ${service}
-	kubectl describe services ${service}
-
-	# Check pods are running
-	cmd="kubectl get pods -o jsonpath='{.items[*].status.phase}' | grep Running"
-	waitForProcess "$wait_time" "$sleep_time" "$cmd"
+	# There appears to be no easy way to formally wait for a loadbalancer service
+	# to become 'ready' - there is no service.status.condition field to wait on.
 
-	# Verify pods IP Addresses
-	IP_ADDRESSES=$(kubectl get pods -o jsonpath='{.items[*].status.podIP}')
+	# Now obtain the local IP:port pair of the loadbalancer service and ensure
+	# we can curl from it, and get the expected result
+	svcip=$(kubectl get service ${service} -o=json | jq '.spec.clusterIP' | sed 's/"//g')
```

Comment:
just wondering if it could be that service is not ready and get an undesired value here?

Reply:
agreed.

Reply:
Yep. as the comment above says though, afaict there is no way to wait on the loadbalancer service - there is no status field to wait on... if you can come up with a not insane way to poll, please paste and I'll put it in.
I resisted just sticking a `sleep` in ;-)

Locally, I have never seen it fail....

Reply:
took a quick look and seems that services are ready right away... and the CI passed correctly. I think we can have it this way.